strawberry_django/fields/base.py

```diff
@@ -43,11 +60,86 @@ def is_basic_field(self) -> bool:
         """
         return False
 
-    @property
-    def type(  # noqa: A003
+    @cached_property
+    def django_type(self) -> type[utils.WithStrawberryDjangoObjectDefinition] | None:
+        origin = self.type
+
+        tdef = get_object_definition(origin)
```

Comment:
nit, can we change tdef to type_definition or object_definition? :D

Reply:
Done! :)